pdal/private/Raster.hpp

```diff
@@ -0,0 +1,172 @@
+* Copyright (c) 2020, Hobu Inc.
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include <pdal/util/ProgramArgs.hpp>
+
+namespace pdal
+{
+
+struct RasterLimits
+{
+public:
+    void addArgs(ProgramArgs& args)
+    {
+         args.add("resolution", "Cell edge size in units of X/Y", edgeLength).setPositional();
+         m_xOriginArg = &args.add("origin_x", "X origin for grid", xOrigin);
+         m_yOriginArg = &args.add("origin_y", "Y origin for grid", yOrigin);
+         m_widthArg = &args.add("width", "Number of cells in the X direction", width);
+         m_heightArg = &args.add("height", "Number of cells in the Y direction", height);
+    }
+
+    int checkArgs()
+    {
+        int args = 0;
+        if (m_xOriginArg->set())
+            args++;
+        if (m_yOriginArg->set())
+            args++;
+        if (m_widthArg->set())
+            args++;
+        if (m_heightArg->set())
+            args++;
+        return args;
+    }
+
+public:
+    double xOrigin;
+    double yOrigin;
+    size_t width;
+    size_t height;
+    double edgeLength;
+
+private:
+    Arg *m_xOriginArg;
+    Arg *m_yOriginArg;
+    Arg *m_widthArg;
+    Arg *m_heightArg;
+};
+
+// This is a raw raster suitable for sticking into GDAL. X goes right to left, Y goes from
```

Comment:
Right to left? This always trips me up, but I thought Y was the axis that typically gets flipped when not careful.

Reply:
Oops.